File: CyclicSort/sortColors.java
//https://leetcode.com/problems/sort-colors/


class Solution {
    public static void sortColors(int[] nums) {
        if (nums.length < 2) {
            return;
        }
        int min = 0;
        int max = nums.length - 1;

        for (int i  = min; i < max; i++) {
            if (nums[i] == 0) {
                int temp = nums[i];
                nums[i] = nums[min];
                nums[min] = temp;
                i++;
                min++;
            }
            else if (nums[i] == 2) {
                int temp = nums[i];
                nums[i] = nums[max];
                nums[max] = temp;
                max--;
            }
            else {
                i++;
            }
        }
        System.out.println(nums); 
    }
}


class Test {
    public static void main(String[] args) {
        int[] array = new int[]{0, 2, 1, 1, 2};
        Solution.sortColors(array);
    }
}

File: CyclicSort/sortColors.py
# https://leetcode.com/problems/sort-colors/

class Solution(object):
    def sortColors(self, nums):
        """
        This function moves all zeroes to the beginning, all twos to the right, and leaves the ones unchanged. 

        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        if nums.length <= 1:
            return
        beginning = 0
        end = nums.length
        
        for

def main():
    solution = Solution()
    print(solution.sortColors([2, 0, 2, 1, 1, 0]))


if __name__ == "__main__":
    main()